plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/groupBy.kt

```diff
@@ -199,4 +205,195 @@ abstract class GroupByAggregator(val defaultName: String) : AbstractSchemaModifi
+            return PluginDataFrameSchema(receiver.keys.columns() + resolvedColumns)
+        } else {
+            val resolvedColumns = columns?.resolve(receiver.keys)?.map { it.column }!!.toList()
+            // TODO: how to handle type of multiple columns
+            val aggregated =
+                makeNullable(simpleColumnOf(name ?: defaultName, (resolvedColumns[0] as SimpleDataColumn).type.type))
+            return PluginDataFrameSchema(receiver.keys.columns() + aggregated)
+        }
+    }
+}
+
+/** Implementation for `sum` */
+class GroupBySum0 : GroupByAggregator3(defaultName = "sum")
+
+/** Implementation for `mean` */
+class GroupByMean0 : GroupByAggregatorMean(defaultName = "mean")
+
+/** Implementation for `median` */
+class GroupByMedian0 : GroupByAggregator3(defaultName = "median")
+
+/** Implementation for `median` */
+class GroupByMin0 : GroupByAggregator3(defaultName = "min")
+
+/** Implementation for `median` */
+class GroupByMax0 : GroupByAggregator3(defaultName = "max")
+
+/** Implementation for `std` */
+class GroupByStd0 : GroupByAggregator3(defaultName = "std")
+
+abstract class GroupByAggregator4() : AbstractSchemaModificationInterpreter() {
+    val Arguments.receiver by groupBy()
+
+    override fun Arguments.interpret(): PluginDataFrameSchema {
+        val resolvedColumns = receiver.groups.columns()
+            .filter {
+                it is SimpleDataColumn
+                    && it.type.type.isSubtypeOf(session.builtinTypes.numberType.type, session)
+            }
+
+        return PluginDataFrameSchema(receiver.keys.columns() + resolvedColumns)
+    }
+}
+
+class GroupBySum1 : GroupByAggregator4()
+
+
+
+class GroupByStd1 : GroupByAggregator4()
+
+class GroupByMean1 : GroupByAggregator4()
+
+private fun ConeKotlinType.isSubtypeOfComparable(session: FirSession): Boolean {
+    val comparableTypes: List<FirImplicitBuiltinTypeRef> = listOf(
+        session.builtinTypes.booleanType,
+        session.builtinTypes.numberType,
+        session.builtinTypes.byteType,
+        session.builtinTypes.shortType,
+        session.builtinTypes.intType,
+        session.builtinTypes.longType,
+        session.builtinTypes.doubleType,
+        session.builtinTypes.floatType,
+        session.builtinTypes.uIntType,
+        session.builtinTypes.charType,
+        session.builtinTypes.stringType
+    )
+
+    return comparableTypes.any { it.type.isSubtypeOf(this, session)  }
+}
+
+ /** class FirImplicitThrowableTypeRef(
+    source: KtSourceElement?
+) : FirImplicitBuiltinTypeRef(source, StandardClassIds.Comparable) */
+
+abstract class GroupByAggregatorComparable() : AbstractSchemaModificationInterpreter() {
+    val Arguments.receiver by groupBy()
+
+    override fun Arguments.interpret(): PluginDataFrameSchema {
+        val resolvedColumns = receiver.groups.columns()
+            .filter {
+                it is SimpleDataColumn
+                    && it.type.type.isSubtypeOfComparable(session)
```

Comment:
@koperagen here, I am trying to check if types if comparable to emulate the interComparable() internal function behaviour for median/min/max functions

Reply:
valuesAreComparable should be impemented like this, to follow `public fun <T : Comparable<T>> Iterable<T>.maxOrNull(): T?` signature. 
```
public fun AnyCol.valuesAreComparable(): Boolean {
    val other = Comparable::class.createType(listOf(KTypeProjection(KVariance.IN, type)))
    return type.isSubtypeOf(other)
}
```

Now it's possible to break it with this code:
```
val i = 1
    val i1: Comparable<Int> = object : Comparable<Int> {
        override fun compareTo(other: Int): Int {
            return other
        }
    }
    val col by columnOf(i, i1)
    dataFrameOf(col).max().print()
```
Because indeed, Comparable<Int> does not implement Comparable<Comparable<Int>>
It's not possible to call `listOf(i, i1).maxOrNull()`

Reply:
So for plugin:
```
fun isInterComparable(col: SimpleDataColumn): Boolean {
      val comparable = StandardClassIds.Comparable.constructClassLikeType(arrayOf(col.type.type))
      return col.type.type.isSubtypeOf(comparable, session)
  }
```

Reply:
@koperagen I created a branch with this suggested fix, but it doesn't produce the expected results either https://github.com/Kotlin/dataframe/tree/values-are-comparable-fix

Reply:
@koperagen actually, looking better at it I think you're right and your fix is too. There's some tests that need to be rewritten.

Reply:
In theory, this should be the plugin version then:
```kt
fun isIntraComparable(col: SimpleDataColumn): Boolean {
    val comparable = StandardClassIds.Comparable.constructClassLikeType(
        typeArguments = arrayOf(col.type.type.withNullability(ConeNullability.NOT_NULL, session.typeContext)),
        isNullable = col.type.type.isNullable,
    )
    return col.type.type.isSubtypeOf(comparable, session)
}
```
(didn't test this yet)